### DISCUSSION TITLE
Is the Powerwall-Dashboard version displayed somewhere within the Dashboard?

### DISCUSSION THREAD

**Discussion by tginther** [jasonacox/Powerwall-Dashboard#541] [Q&A]
Hi Everyone,

I have been running Powerwall-Dashboard since 8/30 and today upgraded to v4.5.5. Because my Gateway updated firmware a couple days ago and I had to change the default password, I decided to see if there was an update to Powerwall-Dashboard, which there was. I updated to v4.5.5 but am curious is there a way to see the current version within the dashboard? I couldn't find anything. The only way I knew to update today is because the releases show v4.5.5 released 2 weeks ago so I knew it was newer than the version I installed on 8/30.

Thanks in advance for your help!

Terry

**Comment by jasonacox** [comment]
Hi @tginther - The version of the Dashboard is in the file `VERSION` or you can run `verify.sh` and it will show the version you are running as well as the version (and status) of all the components. I always thought it would be nice to have it show up in the Grafana dashboard as well, but never spent time to figure out the best way.

BTW, What firmware version are you running?

**Reply by tginther** [reply]
Hi @jasonacox,

I'm running Firmware: 24.36.2. I hadn't looked at the dashboard in a few days and on Tue, opened it up and saw it hadn't been running for a few days. I tried logging into the Tesla Dashboard and of course got the message I needed to change my password, and had to go through those gyrations - they certainly don't make that an easy process!

**Reply by tginther** [reply]
Hi @jasonacox,

I just ran the verify.sh, and it does show the version (4.4.6), but also has some errors which I'm not sure how to interpret. The Dashboard seems to be running just fine (see screenshot). I've attached the output of verify.sh. Do I not have it set up correctly?


<img width="654" alt="Screenshot 2024-10-31 at 2 25 13 PM" src="https://github.com/user-attachments/assets/340a2948-f09f-4322-9423-a3f02e437f20">
<img width="1892" alt="Screenshot 2024-10-31 at 2 25 46 PM" src="https://github.com/user-attachments/assets/c2632a82-95e5-405f-a737-7407eadd6bd4">

**Reply by jasonacox** [reply]
Well, I never tried to run verify.sh with a white background terminal. Can you run it again with a dark mode background? That's white font color is something I need to address. ;)

>  I needed to change my password,

Did you use something like this:

https://github.com/jasonacox/Powerwall-Dashboard/issues/536#issuecomment-2449033170

> Connect to Powerball through TEG WiFi with Password of S + Serial Number
> Open: 192.168.91.1
> Change Customer Passwort, through Lost Passwort
> Turn Powerwall Off and On
> Input last five digits of serial number as old PW
> Use Password at least 8 digits, up / lower / number / @#€_
> Hit change PW

**Reply by tginther** [reply]
I changed the background and saw it wasn't making a connection to the Gateway. Also couldn't ping it. I went into the iPhone app and had to turn on Remote Service Access in order to ping and log into the Gateway on my local network. Once I did that and re-ran verify, everything passed except TEDAPI since I'm using my home WiFi rather than Tesla's WiFI. I'm attaching the verify results and screenshots of the Dashboard. I'm assuming there is no string data available to the Dashboard is because of the TEDAPI not being available? Is there any way to access that API via my home WiFi? Also, it seems I have to set a reminder to re-enable Remote Service Access every 30 days?

<img width="682" alt="Screenshot 2024-11-01 at 10 55 31 AM" src="https://github.com/user-attachments/assets/45903b53-6ab9-4a72-abc8-dbc11ef3b5ef">
<img width="2559" alt="Screenshot 2024-11-01 at 11 23 47 AM" src="https://github.com/user-attachments/assets/dbc69c8d-002c-43d5-bc31-22843c53b00c">
<img width="2557" alt="Screenshot 2024-11-01 at 11 24 11 AM" src="https://github.com/user-attachments/assets/1f51482c-1975-469c-ade5-127d29fab0d3">

**Reply by tginther** [reply]
Hi @jasonacox,

On my AP's I created a broadcast for my TEG-xxx network, and added a static route (per these instructions: https://www.reddit.com/r/TeslaSolar/comments/1d4vpkg/teg_network_access/). I then changed setup to access the Tesla Gateway at 192.168.91.1 and restarted the Powerwall-Dashboard. Verified by logging into the Tesla Gateway via 192.168.91.1 via Safari, which worked with no issues.

verify.sh now reports no errors (see attached), however Powerwall+ String Data still showing no data. Gateway TEDAPI now shows Available, however TEDAPI Vitals reports Not Connected.

Any way to fix this?

<img width="1207" alt="Screenshot 2024-11-01 at 12 56 17 PM" src="https://github.com/user-attachments/assets/1e2f9ab9-4b19-4cc7-8d48-ea59d6525aaf">
<img width="1349" alt="Screenshot 2024-11-01 at 12 56 29 PM" src="https://github.com/user-attachments/assets/03dcf946-0782-4222-ac05-8d002e69e1ea">

**Reply by jasonacox** [reply]
It sounds like you were able to get it to connect to the TEDAPI IP address via your WiFi.  We talk about that process in the extended vitals section of the README and there is a script (`add_route.sh`) that can help: https://github.com/jasonacox/Powerwall-Dashboard?tab=readme-ov-file#extended-device-vitals-metrics-beta

You are not getting a good connection to the TEDAPI gateway because it requires setting the `PW_GW_PWD` variable to your gateway password (usually the same as the WiFi password for the Tesla Gateway) in the `pypowerwall.env` file. You can do that manually and then restart with `./compose-dash.sh up -d` or you can run `setup.sh` and enter the GW_PWD there.

**Reply by hulkster** [reply]
> Hi @tginther - The version of the Dashboard is in the file `VERSION` or you can run `verify.sh` and it will show the version you are running as well as the version (and status) of all the components. I always thought it would be nice to have it show up in the Grafana dashboard as well, but never spent time to figure out the best way.

That would be nifty if easy to display it on the dashboard. 

I would recommend showing version and date ... since this gives you an idea at a glance how "recent" you are.

**Reply by tginther** [reply]
<img width="432" alt="Screenshot 2024-11-04 at 8 30 21 AM" src="https://github.com/user-attachments/assets/d65c0eaa-fa85-42cd-8d6c-3ec2202d1ade">
> You are not getting a good connection to the TEDAPI gateway because it requires setting the `PW_GW_PWD` variable to your gateway password (usually the same as the WiFi password for the Tesla Gateway) in the `pypowerwall.env` file. You can do that manually and then restart with `./compose-dash.sh up -d` or you can run `setup.sh` and enter the GW_PWD there.

I can't get it to work. I manually added the PW_GW_PWD in pypowerwall.env (see attached) and restarted, but Powerwall+ String Data is still showing "No data" on the Dashboard.

**Reply by jasonacox** [reply]
Hi @tginther I would love to see the logs from pypowerwall, would you be able to try this:

```bash
docker stop pypowerwall
docker rm pypowerwall
./compose-dash.sh up -d
docker logs pypowerwall -f
```

**Reply by jeffkruczynski** [reply]
Hi @jasonacox I am running into this same issue (not getting a TEDAPI connection). I ran the steps you just mentioned above and below is the output.
(I have a route to 192.168.91.1 set up via my router and confirmed that I can get to the gateway in the browser at 192.168.12.209)
PW_GW_PWD is set in pypowerwall.env.
It seems odd that `verify.sh` is showing me on 24.36.2, which according to other discussions I'm finding means the vitals API should be supported (I think?), but the log alert below says it isn't.
Any suggestions / fixes to what I'm doing?

![verify output](https://github.com/user-attachments/assets/377d39aa-06f2-449d-be94-ed13965b5b86)


`11/21/2024 03:12:21 PM [proxy] [INFO] pyPowerwall [0.10.9] Proxy Server [t63] - HTTP Port 8675
11/21/2024 03:12:21 PM [proxy] [INFO] pyPowerwall Proxy Started
11/21/2024 03:12:21 PM [proxy] [INFO] pyPowerwall Proxy Server - Local Mode
11/21/2024 03:12:21 PM [proxy] [INFO] Connected to Energy Gateway 192.168.12.209 (My Home)
11/21/2024 03:12:25 PM [pypowerwall.local.pypowerwall_local] [ERROR] 404 Powerwall API not found at https://192.168.12.209/api/devices/vitals
11/21/2024 03:12:25 PM [pypowerwall.local.pypowerwall_local] [ERROR] 404 Powerwall API not found at https://192.168.12.209/api/devices/vitals
11/21/2024 03:12:25 PM [pypowerwall.local.pypowerwall_local] [ERROR] 404 Powerwall API not found at https://192.168.12.209/api/devices/vitals
11/21/2024 03:12:25 PM [pypowerwall.local.pypowerwall_local] [ERROR] 404 Powerwall API not found at https://192.168.12.209/api/devices/vitals
11/21/2024 03:12:25 PM [pypowerwall.local.pypowerwall_local] [ERROR] 404 Powerwall API not found at https://192.168.12.209/api/devices/vitals
11/21/2024 03:12:25 PM [pypowerwall.local.pypowerwall_local] [ERROR] Firmware 243602 detected - Does not support vitals API - disabling.
11/21/2024 03:12:25 PM [pypowerwall.local.pypowerwall_local] [ERROR] Firmware 243602 detected - Does not support vitals API - disabling.
11/21/2024 03:12:25 PM [pypowerwall.local.pypowerwall_local] [ERROR] Firmware 243602 detected - Does not support vitals API - disabling.
11/21/2024 03:12:25 PM [pypowerwall.local.pypowerwall_local] [ERROR] Firmware 243602 detected - Does not support vitals API - disabling.
11/21/2024 03:12:25 PM [pypowerwall.local.pypowerwall_local] [ERROR] Firmware 243602 detected - Does not support vitals API - disabling.
11/21/2024 03:13:05 PM [pypowerwall] [ERROR] Failed to get /api/system_status/grid_status
11/21/2024 03:13:06 PM [urllib3.connectionpool] [WARNING] Connection pool is full, discarding connection: 192.168.12.209. Connection pool size: 15
11/21/2024 03:13:06 PM [urllib3.connectionpool] [WARNING] Connection pool is full, discarding connection: 192.168.12.209. Connection pool size: 15
11/21/2024 03:13:06 PM [urllib3.connectionpool] [WARNING] Connection pool is full, discarding connection: 192.168.12.209. Connection pool size: 15
11/21/2024 03:13:06 PM [urllib3.connectionpool] [WARNING] Connection pool is full, discarding connection: 192.168.12.209. Connection pool size: 15
11/21/2024 03:13:06 PM [urllib3.connectionpool] [WARNING] Connection pool is full, discarding connection: 192.168.12.209. Connection pool size: 15
`

**Reply by jasonacox** [reply]
Hi @jeffkruczynski !  thanks for pasting the logs...

> Local Mode 11/21/2024 03:12:21 PM [proxy] [INFO] Connected to Energy Gateway 192.168.12.209

This is a problem. It is connecting to 192.168.12.209 instead of 192.168.91.1 - You need to re-run setup and use `192.168.91.1` as the address for the Powerwall, or you can edit the pypowerwall.env file and change it manually, then run:

```bash
docker stop pypowerwall
docker rm pypowerwall
./compose-bash.sh up -d
```

**Reply by jeffkruczynski** [reply]
Done — updated, but it's still showing not connected. Log info below.
I am not sure I understand why it seems to be rejecting the password. I've tried the password I use to access 192.168.12.91 via the browser, as well as the password on the gateway sticker itself, but have gotten the same error both times.

`11/21/2024 10:20:35 PM [proxy] [INFO] pyPowerwall [0.10.9] Proxy Server [t63] - HTTP Port 8675
11/21/2024 10:20:35 PM [proxy] [INFO] pyPowerwall Proxy Started
11/21/2024 10:20:36 PM [pypowerwall.tedapi] [ERROR] Access Denied: Check your Gateway Password
11/21/2024 10:20:36 PM [pypowerwall.tedapi] [ERROR] Failed to connect to Powerwall Gateway
11/21/2024 10:20:36 PM [pypowerwall.tedapi] [ERROR] Access Denied: Check your Gateway Password
11/21/2024 10:20:36 PM [proxy] [INFO] pyPowerwall Proxy Server - Local Mode
11/21/2024 10:20:36 PM [proxy] [INFO] Connected to Energy Gateway 192.168.91.1 (My Home)
11/21/2024 10:20:40 PM [pypowerwall.local.pypowerwall_local] [ERROR] 404 Powerwall API not found at https://192.168.91.1/api/devices/vitals
11/21/2024 10:20:40 PM [pypowerwall.local.pypowerwall_local] [ERROR] 404 Powerwall API not found at https://192.168.91.1/api/devices/vitals
11/21/2024 10:20:40 PM [pypowerwall.local.pypowerwall_local] [ERROR] 404 Powerwall API not found at https://192.168.91.1/api/devices/vitals
11/21/2024 10:20:40 PM [pypowerwall.local.pypowerwall_local] [ERROR] 404 Powerwall API not found at https://192.168.91.1/api/devices/vitals
11/21/2024 10:20:40 PM [pypowerwall.local.pypowerwall_local] [ERROR] Firmware 243602 detected - Does not support vitals API - disabling.
11/21/2024 10:20:40 PM [pypowerwall.local.pypowerwall_local] [ERROR] Firmware 243602 detected - Does not support vitals API - disabling.
11/21/2024 10:20:40 PM [pypowerwall.local.pypowerwall_local] [ERROR] 404 Powerwall API not found at https://192.168.91.1/api/devices/vitals
11/21/2024 10:20:40 PM [pypowerwall.local.pypowerwall_local] [ERROR] Firmware 243602 detected - Does not support vitals API - disabling.
11/21/2024 10:20:40 PM [pypowerwall.local.pypowerwall_local] [ERROR] Firmware 243602 detected - Does not support vitals API - disabling.
11/21/2024 10:20:40 PM [pypowerwall.local.pypowerwall_local] [ERROR] Firmware 243602 detected - Does not support vitals API - disabling.`

**Reply by jasonacox** [reply]
Suggestions:
* The password should be the same one you would use to join the Gateway WiFi for installation (the QR code password). Very by using your phone to join the `TEG-xxx` WiFi and that password.
* If that doesn't work, try using just the last 5 characters of the password written on the Powerwall Gateway (someone reported that).

**Reply by jeffkruczynski** [reply]
That's what's odd — I can connect to the WiFi network just fine with my devices, using the last 5 characters of that QR code password, and I can sign into the app at 192.168.91.1 using the same last 5. But I've tried setting PW_GW_PWD to both of them, and neither has worked.

**Reply by jasonacox** [reply]
@jeffkruczynski what version of Powerwall and Powerwall Gateway do you have? 

I'm running out of ideas too, but perhaps you could try the command line mode of pypowerwall to see if that works?

```bash
pip install pypowerwall
python3 -m pypowerwall tedapi
```

**Reply by jeffkruczynski** [reply]
Well...I rebuilt the pypowerwall.env file this morning...and am now connected.
I had double-checked and triple-checked the password when trying this yesterday, but it's working now. Maybe I had it wrong, or had a non-printing character in there, or who knows what I did. But in any event, it's working now. Thanks for the help!

![image](https://github.com/user-attachments/assets/3792fcc1-1999-4357-88ab-fa140544437c)

**Reply by jasonacox** [reply]
That's fantastic!!! 

It would have been interesting to see what the `diff` was between the two version to see what was causing the issue. 
But anyway, how did you rebuild pypowerwall.env?  Did you delete and manually create it, run `setup.sh` again or copy over the pypowerwall.sample.env file? It could help others who see this same thing. I also wonder if there is something I could add to verify.sh that would detect potential issues (e.g. non-printable characters). 

Thanks for the update! Congrats on getting it working.

**Reply by jeffkruczynski** [reply]
I used "rebuild" loosely 🤣. I used the Synology DSM text editor to completely delete the PW_GW_PWD line, and then re-added it and manually typed in the password again, then saved the file. I originally decided to try that because I'd added that line to the end of the file, and even though I know it shouldn't matter, I figured I'd try moving it up before the PW_PASSWORD line, just in case. Once it started working, I moved it down below that line again, just to be sure it wasn't the location in the file. It wasn't.
I _think_ when I dropped the password in the first time, I had copied it from the photo I took of the QR code label. I am pretty sure I reviewed it and it looked right, but maybe it wasn't, or maybe the copy grabbed something funky.
It's possible that updating verify.sh to find non-printing characters could help — if that _was_ the issue and I had to guess, it might've been a CR+LF mix somehow, I don't know.
